tests/integration_tests/test_wandb.py

```diff
@@ -31,79 +32,163 @@ def _multiobjective_func(trial: optuna.trial.Trial) -> Tuple[float, float]:
+    study.optimize(_wrapped_func, n_trials=1, callbacks=[wandbc])
+    wandb.run.config.update.assert_called_once_with(expected)
+
+    wandb.run = None
+    study.enqueue_trial(trial_params)
+    study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
+    wandb.init().config.update.assert_called_once_with(expected)
 
 
 @mock.patch("optuna.integration.wandb.wandb")
 def test_multiobjective_attributes_set_on_epoch(wandb: mock.MagicMock) -> None:
 
-    wandb.config.update = mock.MagicMock()
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+    trial_params = {"x": 1.1, "y": 2.2}
+    expected = {"direction": ["MINIMIZE", "MAXIMIZE"], "x": 1.1, "y": 2.2}
 
-    wandbc = WeightsAndBiasesCallback()
     study = optuna.create_study(directions=["minimize", "maximize"])
+    wandbc = WeightsAndBiasesCallback()
+
+    study.enqueue_trial(trial_params)
     study.optimize(_multiobjective_func, n_trials=1, callbacks=[wandbc])
+    wandb.run.config.update.assert_called_once_with(expected)
 
-    expected = {"direction": ["MINIMIZE", "MAXIMIZE"]}
-    wandb.config.update.assert_called_once_with(expected)
+    wandbc = WeightsAndBiasesCallback(as_multirun=True)
+    wandb.run = mock.MagicMock()
+    _wrapped_func = wandbc.track_in_wandb(_multiobjective_func)
+
+    study.enqueue_trial(trial_params)
+    study.optimize(_wrapped_func, n_trials=1, callbacks=[wandbc])
+    wandb.run.config.update.assert_called_once_with(expected)
+
+    wandb.run = None
+    study.enqueue_trial({"x": 1.1, "y": 2.2})
+    study.optimize(_multiobjective_func, n_trials=1, callbacks=[wandbc])
+    wandb.init().config.update.assert_called_once_with(expected)
 
 
 @mock.patch("optuna.integration.wandb.wandb")
-def test_log_api_call_count(wandb: mock.Mock) -> None:
+def test_log_api_call_count(wandb: mock.MagicMock) -> None:
 
-    wandb.log = mock.MagicMock()
+    wandb.sdk.wandb_run.Run = mock.MagicMock
 
+    study = optuna.create_study()
     wandbc = WeightsAndBiasesCallback()
     target_n_trials = 10
-    study = optuna.create_study()
     study.optimize(_objective_func, n_trials=target_n_trials, callbacks=[wandbc])
-    assert wandb.log.call_count == target_n_trials
+    assert wandb.run.log.call_count == target_n_trials
+
+    wandbc = WeightsAndBiasesCallback(as_multirun=True)
+    wandb.run = mock.MagicMock()
+    _wrapped_func = wandbc.track_in_wandb(_objective_func)
+    study.optimize(_wrapped_func, n_trials=target_n_trials, callbacks=[wandbc])
+
+    assert wandb.run.log.call_count == target_n_trials
+
+    wandb.run = None
+    study.optimize(_objective_func, n_trials=target_n_trials, callbacks=[wandbc])
+    assert wandb.init().log.call_count == target_n_trials
 
 
 @pytest.mark.parametrize(
     "metric,expected", [("value", ["x", "y", "value"]), ("foo", ["x", "y", "foo"])]
 )
 @mock.patch("optuna.integration.wandb.wandb")
-def test_values_registered_on_epoch(wandb: mock.Mock, metric: str, expected: List[str]) -> None:
+def test_values_registered_on_epoch(
```

Comment:
General comments here:
* How about moving decorator tests to separate case  with custom objective function calling `wandb.log`. We should be testing if all metrics (callback + decorator) are registered.
* Remaining logic in this test can be simplified by using `pytest.mark.parametrize` instead of deepcopying a study.